.github/workflows/wheels-pure.yml

```diff
@@ -0,0 +1,144 @@
+    steps:
+    - name: checkout code repo
+      uses: actions/checkout@v3
+      with:
+        fetch-depth: 0 # unshallow fetch for setuptools-scm
+        persist-credentials: false
+
+    - name: Build pure python wheel with pip
+      uses: rapidsai/shared-action-workflows/purewheel@feat/wheel-ci-actions-2
+      with:
+        python-package-cuda-suffix: ${{ inputs.python-package-cuda-suffix }}
+        python-package-versioneer-override: ${{ inputs.python-package-versioneer-override }}
+        package-dir: ${{ inputs.package-dir }}
+        output-dir: dist
+        repo-dir: ./
+
+    - name: Download gha-tools
+      uses: rapidsai/shared-action-workflows/gha-tools@feat/wheel-ci-actions-2
+
+    - name: Upload wheels to downloads.rapids.ai
+      run: rapids-upload-wheels-to-s3 dist
+      env:
+        RAPIDS_PY_WHEEL_NAME: ${{ inputs.package-name }}
+        RAPIDS_BUILD_TYPE: "pull-request"
+        RAPIDS_REPOSITORY: ${{ github.repository }}
+        RAPIDS_SHA: ${{ github.sha }}
+        RAPIDS_REF_NAME: ${{ github.ref_name }}
+        RAPIDS_PY_VERSION: "3.8"
+        AWS_ACCESS_KEY_ID: ${{ secrets.RAPIDSAI_GHA_AWS_ACCESS_KEY_ID }}
+        AWS_SECRET_ACCESS_KEY: ${{ secrets.RAPIDSAI_GHA_AWS_SECRET_ACCESS_KEY }}
+
+  wheel-test:
+    name: wheel test pure
+    runs-on: [self-hosted, linux, amd64, gpu-v100-495-1]
+    needs: wheel-build
+    container:
+      image: rapidsai/manylinux_2_31_x86_64
+      env:
+        NVIDIA_VISIBLE_DEVICES: ${{ env.NVIDIA_VISIBLE_DEVICES }} # GPU jobs must set this container env variable
+        RAPIDS_PY_WHEEL_NAME: ${{ inputs.package-name }}
+        RAPIDS_REPOSITORY: ${{ github.repository }}
+        RAPIDS_SHA: ${{ github.sha }}
+        RAPIDS_REF_NAME: ${{ github.ref_name }}
+        RAPIDS_PY_VERSION: "3.8"
+        RAPIDS_BUILD_TYPE: "pull-request"
+        AWS_ACCESS_KEY_ID: ${{ secrets.RAPIDSAI_GHA_AWS_ACCESS_KEY_ID }}
+        AWS_SECRET_ACCESS_KEY: ${{ secrets.RAPIDSAI_GHA_AWS_SECRET_ACCESS_KEY }}
+        RAPIDS_BEFORE_TEST_COMMANDS_AMD64: ${{ inputs.test-before }}
+        CIBW_TEST_REQUIRES: ${{ inputs.test-requires }}
+        CIBW_TEST_EXTRAS: ${{ inputs.test-extras }}
+        CIBW_TEST_COMMAND: ${{ inputs.test-unittest }}
+        RAPIDS_WHEEL_VER_OVERRIDE: "py3-none-any" # for a pure wheel
+    steps:
+    - name: Run nvidia-smi to make sure GPU is working
+      run: nvidia-smi
+
+    - name: Install private index credentials in cibuildwheel container
+      run: printf 'machine pypi.k8s.rapids.ai\n\tlogin cibuildwheel\n\tpassword ${{ secrets.RAPIDSAI_PYPI_CI_PASSWORD }}\n' > ~/.netrc
+
+    - name: checkout code repo
+      uses: actions/checkout@v3
+      if: ${{ inputs.test-checkout-src == 'true' }}
+      with:
+        fetch-depth: 0 # unshallow fetch for setuptools-scm
+        persist-credentials: false
+
+    - name: Run citestwheel
+      uses: rapidsai/shared-action-workflows/citestwheel@feat/wheel-ci-actions-2
+
+  wheel-publish:
```

Comment:
Consider a `cipublishwheel` (or `cipubwheel`, whatever I feel like at the time) composite action:

1. so that manylinux + pure don't redefine it
2. so that we can centrally change the publish index in one place